cordova-lib/src/cordova/create.js

```diff
@@ -220,8 +220,22 @@ function create(dir, optionalId, optionalName, cfg) {
             }
         }
     })
-    .then(function(import_from_path) {
-
+    .then(function(input_directory) {
+        var import_from_path = input_directory;
+        var templatePkg;
+        
+        //handle package.json present with dirname
+        if (fs.existsSync(path.join(input_directory,'package.json'))){
```

Comment:
Can we omit this check? If there is no `package.json` in template directory we'll catch this in `catch (e)` below.